### PR TITLE
[installer]: configure prebuilds

### LINKED ISSUE #7145: [Installer]: test prebuilds work with Installer deployments
> Prebuilds are an enterprise-level feature. An [enterprise license](https://www.gitpod.io/enterprise-license) in required

The Installer has been shipped but pre-builds were not part of MVP. Investigate where the Installer is with regards to pre-builds and schedule any required work

- [ ] Get prebuilds working
- [ ] Publicly document how to get prebuilds working in a self-hosted installation

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Related Issue(s)

Fixes #7145

## Release Notes

```release-note
```

### CODE REVIEW

**Comment by roboquat** [comment]
@MrSimonEmms: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankeromnes** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankeromnes` in a comment when ready.

Associated issue: *#7145*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/components/server/deployment.go [hunk 1]
```diff
@@ -130,13 +145,16 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
...
 								RunAsUser:  pointer.Int64(31001),
 							},
-							Ports: []corev1.ContainerPort{{
-								Name:          ContainerPortName,
-								ContainerPort: ContainerPort,
-							}, {
-								Name:          PrometheusPortName,
-								ContainerPort: PrometheusPort,
-							}},
+							Ports: []corev1.ContainerPort{
```
Comment (MrSimonEmms) [inline installer/pkg/components/server/deployment.go:148]: This is a tidy-up of code formatting

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jldec** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jldec` in a comment when ready.

Associated issue: *#7145*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** and additionally assign **jankoehnlein** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankoehnlein` in a comment when ready.

Associated issue: *#7145*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/components/server/types.go [hunk 2]
```diff
@@ -130,14 +130,17 @@ type WorkspaceGarbageCollection struct {
...
+	Enabled             bool   `json:"enabled"`
+	AppId               int32  `json:"appId"`
+	AppIdPath           string `json:"appIdPath"`
+	BaseUrl             string `json:"baseUrl"`
+	WebhookSecret       string `json:"webhookSecret"`
+	WebhookSecretPath   string `json:"webhookSecretPath"`
+	AuthProviderId      string `json:"authProviderId"`
+	CertPath            string `json:"certPath"`
+	MarketplaceName     string `json:"marketplaceName"`
+	MarketplaceNamePath string `json:"marketplaceNamePath"`
```
Comment (csweichel) [inline installer/pkg/components/server/types.go:142]: No self-hosted user will integrate with the GitHub market place. For SaaS we are considering to remove this integration as well. Adding this now does not seem sensical.

File: installer/pkg/components/server/deployment.go (hunk 1 above)
Comment (csweichel) [reply installer/pkg/components/server/deployment.go:150]: Thank you for applying this formatting :)

File: installer/pkg/config/v1/config.go [hunk 3]
```diff
@@ -216,6 +218,10 @@ type WorkspaceTemplates struct {
 	Probe      *corev1.Pod `json:"probe"`
 }
 
+type Meta struct {
+	GitHubApp *ObjectRef `json:"githubApp,omitempty"`
```
Comment (csweichel) [inline installer/pkg/config/v1/config.go:222]: How are SH users expected to use this? Are there docs that describe how to enable prebuilds?

Comment (MrSimonEmms) [reply installer/pkg/config/v1/config.go:222]: > Thank you for applying this formatting :)

I know how to make you happy.

> How are SH users expected to use this?

There will be, as part of this PR when it's ready for review

File: installer/pkg/components/server/types.go (hunk 2 above)
Comment (MrSimonEmms) [reply installer/pkg/components/server/types.go:142]: Ok

Comment (jldec) [reply installer/pkg/components/server/types.go:142]: Yeah, to clarify, the GitHub app is required for managing webhooks for prebuilds.
Installing/authorizing the app in GitHub accounts is now integrated with Gitpod projects.

The GitHub marketplace config should be optional (or managed separately).

**Pull request closed without merging**